daemon/execdriver/lxc/lxc_init_linux.go

```diff
@@ -4,19 +4,38 @@ package lxc
 
 import (
 	"fmt"
+	"os"
 
 	"github.com/docker/libcontainer/utils"
+	"github.com/vishvananda/netns"
 )
 
 func finalizeNamespace(args *InitArgs) error {
 	if err := utils.CloseExecFrom(3); err != nil {
 		return err
 	}
+
+	// Make sure you setup the netns before you setup user since once you
+	// setup as a unprivileged used you cannot set into different namespace.
+	if args.NetNsPath != "" {
```

Comment:
@ashahab-altiscale Thanks for the review. LXC driver seems to be failing after several rebases(even though I verified all the functionality manually when I made this change) but test-integration-cli was always passing in my machine(even now) so I was somewhat misled by that. I have fixed the issues to make LXC driver work again. Please take a look when I push the new diffs.